### PR TITLE
fix: correct checkboxes disabled when reentering response in match in…

### PR DESCRIPTION
**Pull request by Silvia-loza**
Related Issue: https://oat-sa.atlassian.net/browse/AUT-885
checkbox is disabled when re-entering response.

Pre-requisite: To have an item created with a Match interaction

SETPS to test:

- Checkout to branch fix/AUT-885/checkboxes-not-available-after-reentering-response
- Open the Match interaction Item & click response
- Choose "Map response" from drop down menu (Response Processing)
- Tick checkbox "Define correct response"
- Define correct response and the score
- Save and preview.
- Close "Preview" window and go back to Response Properties.

**Actual result:** Checkbox 'Define correct response' is ticked, previously defined correct response does not persist, user cannot select any of checkboxes as response.
**Expected result:** Previously defined correct response persists. Checkboxes are not grayed and user can select any of them as response. Checkbox 'Define correct response' is ticked.
Testing env: http://test-silvia.playground.kitchen.it.taocloud.org:41441

### CODE REVIEW

File: views/js/qtiCreator/widgets/interactions/matchInteraction/states/Map.js [hunk 1]
```diff
@@ -18,11 +18,12 @@ define([
 
         //finally, apply defined correct response and response mapping:
         responseWidget.setResponse(interaction, _.values(response.getCorrect()));
-
-        matchInteractionArea
-            .prop('disabled', true)
-            .prop('checked', false)
-            .addClass('disabled');
+            if(!interaction.metaData.responseMappingMode) {
```
Comment (bziondik) [inline views/js/qtiCreator/widgets/interactions/matchInteraction/states/Map.js:21]: Do we have case when `interaction.metaData.responseMappingMode === false`?
if MAP_RESPONSE is used, then this Map widget will be activated
I think we can just remove this code

**Review by bziondik — CHANGES_REQUESTED** [review]

File: views/js/qtiCreator/widgets/interactions/matchInteraction/states/Map.js (hunk 1 above)
Comment (Silvia-loza) [reply views/js/qtiCreator/widgets/interactions/matchInteraction/states/Map.js:21]: I thought the same and coudn't find a case where it was used, but I thought that I might be missing something... 
If you can't find any use for it either I'll remove it.

File: views/js/qtiCreator/widgets/interactions/matchInteraction/states/Map.js [hunk 2]
```diff
@@ -19,11 +19,6 @@ define([
         //finally, apply defined correct response and response mapping:
         responseWidget.setResponse(interaction, _.values(response.getCorrect()));
 
-        matchInteractionArea
-            .prop('disabled', true)
-            .prop('checked', false)
-            .addClass('disabled');
-
```
Comment (404passionFruit) [inline views/js/qtiCreator/widgets/interactions/matchInteraction/states/Map.js:26]: Any idea why it was here in the first place? :)

Comment (Silvia-loza) [reply views/js/qtiCreator/widgets/interactions/matchInteraction/states/Map.js:26]: No idea...  that's why I was hesitant to delete... maybe @habemuscode knows better as it seems that last changes were made by him last year in [this commit](https://github.com/oat-sa/extension-tao-itemqti/commit/debdf3966842b20e3db41348be0ca0bc4e945820#diff-de84230cfd9c4abac34e1eb867bdce7c17b72965fb6640a14af56950b3345a4d) ?

**Review by bziondik — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Review by habemuscode — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Pull request merged**